Frontend/app/problem/display/displayProblem.html

```diff
@@ -72,21 +90,38 @@ <h3 class="panel-title">
             <!-- Only show the buttons on the active bad step -->
             <div class="panel-body">
               <div *ngIf="!stepIsBeingCorrected">
-                <button class="btn btn-primary btn-sm" (click)="showNextBadStep()" *ngIf="badStepIndex==activeBadStepCount">
-                  Continue to Next Step
+                <button class="btn btn-primary btn-sm" (click)="showNextBadStep()"
+                *ngIf="badStepIndex==activeBadStepCount && badStepIndex != step.badStepPath.length-1">
+                  Correct, go on Mentii
                 </button>
-                <button class="btn btn-danger btn-sm" (click)="incorrectBadStep(badStepIndex,badStep)">
-                  This Step is Incorrect
+
+                <!-- Only show this button when we've reached a solution for a bad step -->
+                <button class="btn btn-primary btn-sm" (click)="showNextBadStep()"
+                *ngIf="badStepIndex==activeBadStepCount && badStepIndex == step.badStepPath.length-1">
+                  Confirm solution
+                </button>
+
+                <!-- Only show this button on the non active bad steps -->
+                <a class="lightBadStep btn btn-sm" (click)="incorrectBadStep(badStepIndex,badStep)"
+                *ngIf="badStepIndex != activeBadStepCount">
+                  Something here looks wrong Mentii
```

Comment:
Using `<a>` instead of `<button>` because of some CSS pseudo-classes that apply to links that I couldn't get working for `<button>`. Otherwise after clicking the `<button>` the text turns black and is hard to read